bessctl/commands.py

```diff
@@ -1881,13 +1875,21 @@ def print_delta(tc, delta, name_len):
             print_header(now[tc].timestamp, max_len)
 
             for tc in tcs:
-                print_delta('W%d %s' % (wids[tc], tc),
-                            get_delta(last[tc], now[tc]), max_len)
+                print_delta(now[tc].timestamp, 'W{} {}'.format(wids[tc], tc),
+                            get_delta(last[tc], now[tc]), max_len, csv)
 
             print_footer(max_len)
 
             for tc in tcs:
                 last[tc] = now[tc]
+    try:
+        csv_path = os.getenv('CSV', None)
+        csv_f = open(csv_path, 'w+') if csv_path is not None else None
```

Comment:
Why open for `w+`?

In any case this is better as a `with` to make sure the file is closed even on interrupt, but `with` is slightly hard to use here since `csv_f` might be None.  You can use a no-op context manager: https://stackoverflow.com/questions/28680442/python-context-for-file-or-none

Reply:
Nice! Implemented this. Much cleaner/safer now-- thanks!